ouroboros-consensus/docs/ReleaseProcess.md

```diff
@@ -0,0 +1,140 @@
+# How Consensus Makes Releases
+
+This document explains how the Consensus team uses Pull Requests (PRs), version numbers, and branches to prepare releases of our packages.
+
+Let us assume our repository contains two packages FOO and BAR.
+Two is enough to explain the complexities.
+The resulting rules can be generalized to any number of packages, as explained at the end of the document.
+
+*Remark*.
+If you disagree with the decisions in this document, please consider reviewing our brainstorming notes and discussions, available at [this document's PR](https://github.com/input-output-hk/ouroboros-network/pull/4207).
+Even if you choose not to do so, please share your concern with us.
+
+## Notation
+
+In this document, we will assume each version number is a MAJOR.MINOR.PATCH triple.
+This is easy to generalize to other conventional notations, such as [the PVP](https://pvp.haskell.org/)'s MAJOR.MAJOR.MINOR.PATCH quadruples.
+
+We refer throughout this document to _the main branch_ (aka `master`, aka `main`).
+
+If you search for "RULE:" in this document, you'll find the key statements.
+
+## Rules for Branches
+
+The Consensus team will sometimes be maintaining multiple MAJOR.MINOR versions of the same package simultaneously.
+EG We might release FOO-2.4.1 on Monday, release FOO-2.3.7 on Wednesday, and release FOO-2.4.2 on Friday, etc.
+
+- Whenever we're maintaining only the one greatest-ever MAJOR.MINOR version of a package, then all work on that package will happen on the main branch.
+- Whenever we're also maintaining some lesser MAJOR.MINOR versions of a package, then some work for that package will also be happening on the respective _release branches_.
+
+A _release branch_ is a branch dedicated to the maintenance of some older release of a package.
+
+RULE: If we have released a package FOO-X.Y.Z, then the release-FOO-X.Y.x branch MUST exist and its first-parent history MUST include the commit released as FOO-X.Y.Z.
```

Comment:
It was not entirely clear to me from reading up to this point why the branch is named release-FOO-X.Y.**x**, though it is explained later. May be good to mention here what **x** represents (I think just a literal `x` character?) or to refer to the later section where it is explained

Reply:
Thanks; yeah, you and Damian both pointed that out. I'll add some concrete examples making that clear.